web/blueprints/mpskclient/__init__.py

```diff
@@ -0,0 +1,158 @@
+    user = get_user_or_404(request.args.get("user_id", default=None, type=int))
+    form = WiFiInterfaceForm(owner_id=user.id)
+
+    def default_response() -> ResponseReturnValue:
+        form_args = {"form": form, "cancel_to": url_for("user.user_show", user_id=user.id)}
+
+        return render_template(
+            "generic_form.html", page_title="MPSK Client erstellen", form_args=form_args, form=form
+        )
+
+    if not form.is_submitted():
+        return default_response()
+        refill_room_data(form, user.room)
+
+    if not form.validate_on_submit():
+        return default_response()
+
+    # existence verified by validator
+    # TODO can't we provide an attribute for that on the form?
+    owner = session.session.get(User, form.owner_id.data)
+    with abort_on_error(default_response), session.session.begin_nested():
+
+        host = lib_mpsk.mpsk_client_create(
+            owner, form.name.data, form.mac.data, processor=current_user
+        )
+    session.session.commit()
+
+    flash("MPSK Client erfolgreich erstellt.", "success")
+    return redirect(url_for("user.user_show", user_id=host.owner.id, _anchor="mpsks"))
+
+
+@bp.route("/<int:mpsk_id>/delete", methods=["GET", "POST"])
+@access.require("hosts_change")
+def mpsk_delete(mpsk_id: int) -> ResponseReturnValue:
+    mpsk = get_mpsk_client_or_404(mpsk_id)
+    owner = mpsk.owner
+    form = FlaskForm()
+
+    def default_response() -> ResponseReturnValue:
+        form_args = {
+            "form": form,
+            "cancel_to": url_for("user.user_show", user_id=owner.id),
+            "submit_text": "Löschen",
+            "actions_offset": 0,
+        }
+
+        return render_template(
+            "generic_form.html", page_title="MPSK Client löschen", form_args=form_args, form=form
+        )
+
+    if not form.is_submitted():
+        return default_response()
+
+    with abort_on_error(default_response), session.session.begin_nested():
+        lib_mpsk.mpsk_delete(mpsk, current_user)
+    session.session.commit()
+
+    flash("MPSK Client erfolgreich gelöscht.", "success")
+    return redirect(url_for("user.user_show", user_id=owner.id, _anchor="mpsks"))
+
+
+@bp.route("/<int:mpsk_id>/edit", methods=["GET", "POST"])
+@access.require("hosts_change")
+def mpsk_edit(mpsk_id: int) -> ResponseReturnValue:
+    mpsk = get_mpsk_client_or_404(mpsk_id)
+
+    form = WiFiInterfaceForm(obj=mpsk)
+
+    def default_response() -> ResponseReturnValue:
+        form_args = {"form": form, "cancel_to": url_for("user.user_show", user_id=mpsk.owner_id)}
+
+        return render_template(
+            "generic_form.html", page_title="MPSK Client editieren", form_args=form_args
+        )
+
+    if not form.is_submitted() or not form.validate():
+        return default_response()
+
+    with abort_on_error(default_response), session.session.begin_nested():
+        lib_mpsk.mpsk_edit(mpsk, mpsk.owner, form.name.data, form.mac.data, current_user)
```

Comment:
btw, this should be a bug right now, precisely because you don't `session.add(client)` in the `mpsk_edit` definition. See the comment on there.

Reply:
I added an session add so that this should be working